pq-crypto/kyber_r3/kyber512r3_cbd.c

```diff
@@ -2,8 +2,8 @@
 #include "kyber512r3_params.h"
 #include "kyber512r3_cbd.h"
 
-#if S2N_ANY_NONPORTABLE_OPTIMIZATIONS_ENABLED
-#error "Compiling portable code with non-portable assembly optimizations is not allowed"
+#if S2N_BLOCK_NONPORTABLE_OPTIMIZATIONS && S2N_ANY_NONPORTABLE_OPTIMIZATIONS_ENABLED
```

Comment:
With this logic getting a bit more complicated it might be better to move it all into a new macro

```c
#if S2N_BLOCK_NONPORTABLE_OPTIMIZATIONS && S2N_ANY_NONPORTABLE_OPTIMIZATIONS_ENABLED
#define S2N_ENSURE_PORTABLE_OPTIMIZATIONS \
    #error "Compiling portable code with non-portable assembly optimizations. This can result in runtime crashes if artifacts are deployed to older CPU's without these CPU instructions"
#else
#define S2N_ENSURE_PORTABLE_OPTIMIZATIONS 
#endif